src/Hl7.Fhir.Support.Poco/Introspection/PropertyMapping.cs

```diff
@@ -204,7 +203,7 @@ private static bool isPrimitiveValueElement(FhirElementAttribute valueElementAtt
 
         }
 
-        private static string buildQualifiedPropName(PropertyInfo p) => p.DeclaringType.Name + "." + p.Name;
+        private static string buildQualifiedPropName(PropertyInfo p) => p.DeclaringType?.Name + "." + p.Name;
```

Comment:
Actually, if this is null, we have a problem and should throw.

Reply:
I throw now an `ArgumentNullException`.